Skip the optimisation for * for only if the next character is
? and [, rather than ? and anything but [.

Reported-by: Reilly Brogan <reilly@reillybrogan.com>
Fixes: e878137f63e6 ("expand: Move stop array closer to strpbrk call")
Signed-off-by: Herbert Xu <herbert@gondor.apana.org.au>

diff --git a/src/expand.c b/src/expand.c
index 992cbe8..1fab394 100644
--- a/src/expand.c
+++ b/src/expand.c
@@ -1911,7 +1911,7 @@ static int pmatch(char *pattern, const char *string)
 				c = *++p;
 			if (!c)
 				return 1;
-			if (c == '?' || c != '[')
+			if (c == '?' || c == '[')
 				c = CTLESC;
 			for (;;) {
 				if (c != CTLESC) {
